e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -667,6 +648,318 @@ private static PrivateKeyFile GetPrivateSshKey(IIoTPlatformTestContext context)
+
+                if (!itemDictionary.ContainsKey("discoveryUrls")
+                    || !itemDictionary.ContainsKey("applicationId")) {
+                    continue;
+                }
+
+                var discoveryUrls = (List<object>)item.discoveryUrls;
+                var itemUrl = (string)discoveryUrls?.FirstOrDefault(url => IsUrlStringsEqual(url as string, context.DiscoveryUrl));
+
+                if (itemUrl != null) {
+                    return item.applicationId;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Unregisters a server identified by <paramref name="applicationId"/>
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        /// <param name="applicationId">Application ID identifying the server to unregister</param>
+        /// <param name="ct">Cancellation token</param>
+        public static async Task Registry_UnregisterServerAsync(
+                IIoTPlatformTestContext context,
+                string applicationId,
+                CancellationToken ct = default) {
+
+            if (string.IsNullOrEmpty(applicationId)) {
+                context.OutputHelper.WriteLine($"{nameof(applicationId)} is null or empty");
+                throw new ArgumentNullException(nameof(applicationId));
+            }
+
+            var accessToken = await GetTokenAsync(context, ct).ConfigureAwait(false);
+
+            var client = new RestClient(context.IIoTPlatformConfigHubConfig.BaseUrl) { Timeout = TestConstants.DefaultTimeoutInMilliseconds };
+
+            var request = new RestRequest(Method.DELETE);
+            request.AddHeader(TestConstants.HttpHeaderNames.Authorization, accessToken);
+            request.Resource = string.Format(TestConstants.APIRoutes.RegistryApplicationsWithApplicationIdFormat, applicationId);
+
+            var response = await client.ExecuteAsync(request, ct).ConfigureAwait(false);
+            Assert.NotNull(response);
+
+            if (!response.IsSuccessful) {
+                context.OutputHelper.WriteLine($"StatusCode: {response.StatusCode}");
+                context.OutputHelper.WriteLine($"ErrorMessage: {response.ErrorMessage}");
+                Assert.True(response.IsSuccessful, "DELETE /registry/v2/application/{applicationId} failed!");
+            }
+        }
+
+        /// <summary>
+        /// Activates (and waits for activated and connected state) the endpoint from <paramref name="context"/>
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        /// <param name="ct">Cancellation token</param>
+        public static async Task Registry_ActivateEndpointAsync(IIoTPlatformTestContext context, CancellationToken ct = default) {
+            var accessToken = await GetTokenAsync(context, ct).ConfigureAwait(false);
```

Comment:
I believe this method needs the endpoint Id to be passed as parameter. Having a single endpointId pet test context limits dramatically E2E testing capabilities.